licenses/types.go

```diff
@@ -0,0 +1,329 @@
+package licenses
+
+// Type identifies a class of software license.
+type Type string
+
+// License types
+const (
+	// Unknown license type.
+	Unknown = Type("")
+	// Restricted licenses require mandatory source distribution if we ship a
+	// product that includes third-party code protected by such a license.
+	Restricted = Type("restricted")
+	// Reciprocal licenses allow usage of software made available under such
+	// licenses freely in *unmodified* form. If the third-party source code is
+	// modified in any way these modifications to the original third-party
+	// source code must be made available.
+	Reciprocal = Type("reciprocal")
+	// Notice licenses contain few restrictions, allowing original or modified
+	// third-party software to be shipped in any product without endangering or
+	// encumbering our source code. All of the licenses in this category do,
+	// however, have an "original Copyright notice" or "advertising clause",
+	// wherein any external distributions must include the notice or clause
+	// specified in the license.
+	Notice = Type("notice")
+	// Permissive licenses are even more lenient than a 'notice' license.
+	// Not even a copyright notice is required for license compliance.
+	Permissive = Type("permissive")
+	// Unencumbered covers licenses that basically declare that the code is "free for any use".
+	Unencumbered = Type("unencumbered")
+
+	ByExceptionOnly = Type("byExceptionOnly")
+	// Forbidden licenses are forbidden to be used.
+	Forbidden = Type("forbidden")
+)
+
+func (t Type) String() string {
+	switch t {
+	case Unknown:
+		// licenseclassifier uses an empty string to indicate an unknown license
+		// type, which is unclear to users when printed as a string.
+		return "unknown"
+	default:
+		return string(t)
+	}
+}
+
+var typeMap = map[string]Type{
```

Comment:
These license types are not available upstream anymore in v2 (I think).
The `TestTypes` test makes sure we provide a mapping for all license names that we can detect.